sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -615,8 +628,9 @@ class kernel_bundle_impl {
       const std::shared_ptr<device_image_impl> &DevImgImpl =
           getSyclObjImpl(DevImg);
       if (std::shared_ptr<kernel_impl> PotentialKernelImpl =
-              DevImgImpl->tryGetSourceBasedKernel(Name, MContext, Self,
-                                                  DevImgImpl))
+              // move is performed only when SourceBasedKernel is not null
+          DevImgImpl->tryGetSourceBasedKernel(Name, MContext, std::move(Self),
```

Comment:
And somewhat replying to my question above, I think it would make sense to modify `tryGetSourceBasedKernel` to accept raw reference as part of this PR. We'd be able to avoid the ugly `Self` that way.

Also, the current version is buggy - if we call these more than once during the loop then the second call is trying to `std::move` something that's already is in the "moved-out-from" state.